### PR TITLE
[libtcod] Update to 1.22.1

### PR DESCRIPTION
**Pull request by HexDecimal**
**Describe the pull request**

- #### What does your PR fix?
  Updates to the latest version.  Adds the `png` and `zlib` features, default on.
  
  In addition default features were missing from the previous port and are now added (I likely got confused by looking at the libtcod project manifest instead of the port manifest.)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

File: versions/l-/libtcod.json [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "08ac6efab8aa18d3c8e4bde44914495640e451a9",
+      "version": "1.22.1",
+      "port-version": 0
+    },
+    {
+      "git-tree": "a9e4ce956ee6a7f71bfd92473e0087b93d9f35dc",
+      "version": "1.22.0",
+      "port-version": 0
+    },
```
Comment (JonLiu1993) [inline versions/l-/libtcod.json:12]:
```suggestion
```

Comment (JonLiu1993) [reply versions/l-/libtcod.json:12]: @HexDecimal,  Thanks foe your pr, we only add one `git-tree` when we merged

**Review by JonLiu1993 — DISMISSED** [review]

File: versions/l-/libtcod.json (hunk 1 above)
Comment (HexDecimal) [reply versions/l-/libtcod.json]: Sorry about that.  I've committed the suggestion.

**Comment by JonLiu1993** [comment]
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Comment by JavierMatosD** [comment]
Thank you!

**Pull request merged**